include/openssl/bn.h

```diff
@@ -507,9 +507,12 @@ const BIGNUM *BN_get0_nist_prime_521(void);
 int (*BN_nist_mod_func(const BIGNUM *p)) (BIGNUM *r, const BIGNUM *a,
                                           const BIGNUM *field, BN_CTX *ctx);
 
-int BN_generate_dsa_nonce(BIGNUM *out, const BIGNUM *range,
+int BN_generate_dsa_random_nonce(BIGNUM *out, const BIGNUM *range,
                           const BIGNUM *priv, const unsigned char *message,
                           size_t message_len, BN_CTX *ctx);
+int BN_generate_dsa_deterministic_nonce(BIGNUM* out, const BIGNUM* range,
+    const BIGNUM* priv, const unsigned char* message,
+    size_t message_len, int hash_type, BN_CTX* ctx);
```

Comment:
I'm not sure we want to add a new public function for this. Like already pointed out, use some control function to change the behaviour of an other API.

Reply:
If you don't want this to be a public function, where should I place the declaration?
in crypto/include/internal/bn_int.h or somewhere else?